File: app/src/main/java/in/completecourse/questionbank/ScanActivity.java
package in.completecourse.questionbank;

import android.content.Intent;
import android.os.Bundle;
import android.util.SparseArray;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import com.google.android.gms.vision.barcode.Barcode;
import java.util.List;
import info.androidhive.barcode.BarcodeReader;

public class ScanActivity extends AppCompatActivity implements BarcodeReader.BarcodeReaderListener{

    private BarcodeReader  barcodeReader;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_scan);
        setTitle("Scan QR Code");

        barcodeReader = (BarcodeReader) getSupportFragmentManager().findFragmentById(R.id.barcode_scanner);
    }

    @Override
    public void onScanned(Barcode barcode) {
        // playing barcode reader beep sound
        barcodeReader.playBeep();
        // ticket details activity by passing barcode
        Intent intent = new Intent(ScanActivity.this, MainActivity.class);
        intent.putExtra("url", barcode.displayValue);
        setResult(100,intent);
        finish();
    }

    @Override
    public void onScannedMultiple(List<Barcode> barcodes) {

    }

    @Override
    public void onBitmapScanned(SparseArray<Barcode> sparseArray) {

    }

    @Override
    public void onScanError(String errorMessage) {

    }

    @Override
    public void onCameraPermissionDenied() {

    }
}

File: app/src/main/java/in/completecourse/questionbank/ScanActivity.kt
package `in`.completecourse.questionbank

import android.content.Intent
import android.os.Bundle
import android.util.SparseArray
import androidx.appcompat.app.AppCompatActivity
import com.google.android.gms.vision.barcode.Barcode
import info.androidhive.barcode.BarcodeReader
import info.androidhive.barcode.BarcodeReader.BarcodeReaderListener

class ScanActivity : AppCompatActivity(), BarcodeReaderListener {
    private var barcodeReader: BarcodeReader? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_scan)
        title = "Scan QR Code"
        barcodeReader = supportFragmentManager.findFragmentById(R.id.barcode_scanner) as BarcodeReader?
    }

    override fun onScanned(barcode: Barcode) {
        // playing barcode reader beep sound
        barcodeReader!!.playBeep()
        // ticket details activity by passing barcode
        val intent = Intent(this@ScanActivity, MainActivity::class.java)
        intent.putExtra("url", barcode.displayValue)
        setResult(100, intent)
        finish()
    }

    override fun onScannedMultiple(barcodes: List<Barcode>) {}
    override fun onBitmapScanned(sparseArray: SparseArray<Barcode>) {}
    override fun onScanError(errorMessage: String) {}
    override fun onCameraPermissionDenied() {}
}